[ui] Remove datasets from a project

evamillan opened the pull request:
This PR adds the option to delete all of a project's datasets that share a URI on the `DatasourceList` component at the project view `/ecosystem/id/project/name`, or delete one specific dataset from the `DatasetTable` on the datasource view `/project/id/datasource/uri`.

Closes #96.

evamillan committed:
[ui] Delete dataset from table

Adds a button to DatasetTable to allow users to remove
a dataset from a project. After deleting one, if there
are no more datasets with the chosen URI, it redirects
to the project view.

Signed-off-by: Eva Millán <evamillan@bitergia.com>

evamillan committed:
[ui] Remove all datasets with a URI from project

Adds a button to delete all the datasets with the same
URI on DatasourceList.

Signed-off-by: Eva Millán <evamillan@bitergia.com>

jjmerchante commented:
Just one detail, if I try to delete a dataset, it shows a dialog. If I click outside the dialog (not to cancel or continue), and I do a different action (like clicking in a project), the dialog is shown again.

I just noticed it happens with every dialog. Is this the expected behavior?

evamillan committed:
[ui] Clear dialog information when clicking outside

Clears the dialog information on the store when
SimpleDialog is closed by clicking on the background
overlay.

Signed-off-by: Eva Millán <evamillan@bitergia.com>

evamillan commented:
It wasn't the intended behavior, thanks for noticing and letting me know! I've updated the PR and it should be fixed now.

jjmerchante commented:
LGTM
